src/app/waste-water/explore-opportunities/explore-opportunities.component.ts

```diff
@@ -22,11 +22,14 @@ export class ExploreOpportunitiesComponent implements OnInit {
   constructor(private wasteWaterService: WasteWaterService) { }
 
   ngOnInit(): void {
-    this.selectedModificationIdSub = this.wasteWaterService.selectedModificationId.subscribe(val => {
-      if (val) {
-        this.modificationExists = true;
-      } else {
-        this.modificationExists = false;
+    this.selectedModificationIdSub = this.wasteWaterService.selectedModificationId.subscribe(selectedModificationId => {
+      if (selectedModificationId) {
+        let modification: WasteWaterData = this.wasteWaterService.getModificationFromId();
+        if (modification) {
+          this.modificationExists = true;
```

Comment:
small change to simplify - Move this.checkExploreOpps() call to this block (it only needs to happen here). Create a param for modification: WasteWaterData and pass ln 28 'modification' to it. Remove the modificationExists check in the checkExoloreOpps().